java/src/main/java/com/logicalclocks/hsfs/engine/HudiEngine.java

```diff
@@ -155,24 +160,27 @@ private FeatureGroupCommit getLastCommitMetadata(SparkSession sparkSession, Stri
 
     // primary keys
     Seq<String> primaryColumns = utils.getPrimaryColumns(featureGroup);
-    if (primaryColumns.isEmpty()) {
-      throw new FeatureStoreException("For time travel enabled feature groups You must provide at least 1 primary key");
-    }
     hudiArgs.put(HUDI_RECORD_KEY, primaryColumns.mkString(","));
 
+    // table name
+    String tableName = utils.getFgName(featureGroup);
+    hudiArgs.put(HUDI_TABLE_NAME, tableName);
+
     // partition keys
     Seq<String> partitionColumns = utils.getPartitionColumns(featureGroup);
     if (!partitionColumns.isEmpty()) {
       hudiArgs.put(HUDI_PARTITION_FIELD, partitionColumns.mkString(":SIMPLE,") + ":SIMPLE");
-      // For precombine key take 1st primary key
-      hudiArgs.put(HUDI_PRECOMBINE_FIELD, primaryColumns.head());
       hudiArgs.put(HUDI_HIVE_SYNC_PARTITION_FIELDS, partitionColumns.mkString(","));
+      hudiArgs.put(HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY, DEFAULT_HIVE_PARTITION_EXTRACTOR_CLASS_OPT_VAL);
+    } else {
+      hudiArgs.put(HUDI_PARTITION_FIELD, "");
+      hudiArgs.put(HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY, HIVE_NON_PARTITION_EXTRACTOR_CLASS_OPT_VAL);
     }
 
-    // table name
-    String tableName = utils.getFgName(featureGroup);
-    hudiArgs.put(HUDI_TABLE_NAME, tableName);
-    hudiArgs.put(HIVE_PARTITION_EXTRACTOR_CLASS_OPT_KEY, DEFAULT_HIVE_PARTITION_EXTRACTOR_CLASS_OPT_VAL);
+    List<Feature> precombineKey = featureGroup.getFeatures().stream().filter(Feature::getHudiPrecombineKey)
```

Comment:
Why is this a list now? In the rest of the code precombine is a single column, Just use getFirst on the stream, no point in building a list when you know you'll have a single element